keras/backend/torch/nn.py

```diff
@@ -686,3 +686,42 @@ def moments(x, axes, keepdims=False, synchronized=False):
         mean = cast(mean, ori_dtype)
         variance = cast(variance, ori_dtype)
     return mean, variance
+
+
+def batch_norm(x, mean, variance, axis, offset=None, scale=None, epsilon=1e-3):
+    x = convert_to_tensor(x)
+    mean = convert_to_tensor(mean).detach()
+    variance = convert_to_tensor(variance).detach()
+    if offset is not None:
+        offset = convert_to_tensor(offset)
+    if scale is not None:
+        scale = convert_to_tensor(scale)
+
+    def _batch_norm():
+        return tnn.batch_norm(
+            input=x,
+            running_mean=mean,
+            running_var=variance,
+            weight=scale,
+            bias=offset,
+            training=False,
+            eps=epsilon,
+        )
+
+    if axis == 1:
+        return _batch_norm()
+
+    if axis < 0:
+        axis = len(x.shape) + axis
+
+    order = list(range(len(x.shape)))
+    order.pop(axis)
+    order.insert(1, axis)
+    x = x.permute(order)
```

Comment:
It sounds to me that permute + native BN op + permute will be more expensive memory wise than doing a low level BN implementation like before. Permutation is expensive.

Reply:
According to the [torch docs](https://pytorch.org/docs/stable/generated/torch.permute.html), the permute is a view of the original tensor. I guess it is not that expensive.

Reply:
If set to use channels_first, the permute would not run as the function returns before reaching it.